integration/hive/hive-openlineage-hook/src/main/java/io/openlineage/hive/client/EventEmitter.java

```diff
@@ -0,0 +1,60 @@
+/*
+/* Copyright 2018-2025 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+package io.openlineage.hive.client;
+
+import static com.google.common.hash.Hashing.sha512;
+
+import io.openlineage.client.Clients;
+import io.openlineage.client.OpenLineage.RunEvent;
+import io.openlineage.client.OpenLineageClient;
+import io.openlineage.client.OpenLineageClientException;
+import io.openlineage.client.OpenLineageClientUtils;
+import io.openlineage.hive.api.OpenLineageContext;
+import io.openlineage.hive.util.NetworkUtils;
+import java.util.UUID;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.hadoop.conf.Configuration;
+
+@Getter
+@Slf4j
+public class EventEmitter implements AutoCloseable {
+  private final OpenLineageClient client;
+  private final UUID runId;
+  private final String jobName;
+  private final String jobNamespace;
+
+  public EventEmitter(OpenLineageContext olContext) {
+    Configuration conf = olContext.getHadoopConf();
+    this.client = Clients.newClient(olContext.getOpenLineageConfig());
+    this.runId = UUID.randomUUID();
+    this.jobNamespace =
+        conf.get(
+            HiveOpenLineageConfigParser.NAMESPACE_KEY, getJobNamespace(olContext.getQueryString()));
+    this.jobName =
+        conf.get(
+            HiveOpenLineageConfigParser.JOB_NAME_KEY, NetworkUtils.LOCAL_IP_ADDRESS.getHostName());
```

Comment:
Does it mean that all jobName values produced by this integration will be static?
Is it possible to get job name from user-specified variables, like this?
```sql
SET openlineage.job.name='myjob';
SET hivevar:openlineage.job.name='myjob';
```
https://stackoverflow.com/questions/12464636/how-to-set-variables-in-hive-scripts

Reply:
you can get the user specified name and namespace with
```
SET hive.openlineage.job.name=myjob;
SET hive.openlineage.namespace=my-namespace;
```